edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/routing/RoutingEdgeHub.cs

```diff
@@ -57,9 +57,13 @@ public Task ProcessDeviceMessage(IIdentity identity, IMessage message)
 
         public Task ProcessDeviceMessageBatch(IIdentity identity, IEnumerable<IMessage> messages)
         {
-            IEnumerable<IRoutingMessage> routingMessages = Preconditions.CheckNotNull(messages)
-                .Select(m => this.ProcessMessageInternal(m, true));
-            return this.router.RouteAsync(routingMessages);
+            Metrics.MessageCount(identity, messages.Count());
+            using (Metrics.MessageLatency(identity))
+            {
+                IEnumerable<IRoutingMessage> routingMessages = Preconditions.CheckNotNull(messages)
```

Comment:
Move the Preconditions check to the top of the method. Also, can you pls add nameof(messages) parameter to it..

Reply:
Done.